JSON syntax and OpenAPI spec fixes, to silence (some) swagger/openapi codegen errors.

tesueret-msft opened the pull request:
While I'm tilting at the windmill of codegen to consume Sentry's public API, I noted some easy-to-fix JSON and OpenAPI validation errors in `api-docs`.

tesueret-msft commented on api-docs/components/schemas/key.json:
```diff
@@ -1,75 +1,77 @@
-                "choices": {
-                    "type": "array",
-                    "items": {
+{
```
Pay no attention to most of this diff; encapsulating the schema object with `{}` is the only real change.

tesueret-msft commented on api-docs/paths/scim/member_index.json:
```diff
@@ -134,7 +134,7 @@
                             },
                             "groups": {
-                                "type": "array"
+                                "$ref": "../../components/schemas/scim/group_list.json#/GroupList"
```
This was a wild guess, I haven't actually checked the API code -- corrections appreciated 🥳

manuzope commented:
Thanks for the PR, @tesueret-msft! I'll get it scheduled to be reviewed. We don't have a firm timeline on when this will be reviewed but wanted to let you know that it's on our radar!

manuzope commented:
Hey @tesueret-msft, what are you using that gives these validation errors? Want to try running it too.

tesueret-msft commented:
Hey @manuzope !  If I recall correctly, I tried both openapi-generator and swagger-codegen, but the first is what I have handy for a repro now (easiest to setup with npm):

```
npm install -g @openapitools/openapi-generator-cli
mkdir sentry-openapi-codegen
openapi-generator-cli generate -i api-docs/openapi.json -g csharp -o sentry-openapi-codegen
```

This should start to show the errors I tackled in these PR changes.

AniketDas-Tekky commented:
Hi, I've been looking at this PR the last few days. I managed to get the openapi generator working (i think). I have a couple of requests:
1) Do you mind sharing what your output looks like. I want to make sure we're seeing the same errors.
2) Could you rebase this PR with top-of-tree? We made some changes to our docs recently.

Thanks

tesueret-msft commented:
Hi @AniketDas-Tekky , thanks for taking a look!

I've pushed a freshly-merged branch, no worrying conflicts (one of the SCIM spec files I changed is now deleted).

Attached is a copy of the openapi-generator-cli errors I see (as of today) on the sentry `master` copy.  This isn't a comprehensive list of errors I walked through back in September; once those logged errors are fixed, the generator will get further and find more.

... I might also note that the C# code this currently generates isn't correct, yet, but I want to approach that one step at a time.  :D

[sentry-openapi-codegen.log](https://github.com/getsentry/sentry/files/8454597/sentry-openapi-codegen.log)

JoshFerge commented:
/gcbrun

JoshFerge commented:
thanks for the PR @tesueret-msft! For building your client, you may want to use our dereferenced JSON, which is located here: https://github.com/getsentry/sentry-api-schema. I've had better experiences overall with that.
